### PR TITLE
feat(settings): setting dialog

### PR DESCRIPTION
**Pull request by virtuacoplenny**
- Move device selection, profile edit, language select, moderator
  options, and server auth into one modal with tabs.
- Remove side panel profile and settings and logic used to update
  them.
- Pipe server auth status into redux to display in the settings
  dialog.
- Change filmstrip only device selection popup to use the new
  stateless settings dialog component.

### CODE REVIEW

**Comment by virtuacoplenny** [comment]
The relationship between components needs a better look. I bet there's a good way to simplify the passing around of props that I'm not seeing right now. However, I need to make this PR early so I can start finding out what tests fail.

**Comment by saghul** [comment]
General observation, since you asked: I'd have each tab pull the state they need from redux instead of having the dialog have all the props and callbacks required.

Have a look at the mobile Toolbox: the Toolbox component is pretty dumb, all it knows is to render the children, aka the buttons. Then each button is responsible for knowing what to do.

I think applying the same principle here could help. (these are my 2 cents)

**Comment by virtuacoplenny** [comment]
One issue here is I want to support cancelling of changes. So the user can select new devices in the dropdown, change display name, check some boxes, press cancel, and those changes won't be saved. I'm not sure how I would implement such with each tab pulling its own state. Maybe the tabs pull their own state but also the dialog itself pulls state so it know what settings have changed?

**Comment by virtuacoplenny** [comment]
I ended up picking parts of the advice from Saul, Zoli, and Hristo and changing the relationship so there is a non-connected SettingsModal which creates each tab and passes them into Atlaskit Tabs. Full web wraps SettingsModal around a component that is connected to redux. Filmstrip only device selection wraps SettingsModal around a component that uses the transport API to communicate changes to the main window. The end result two removed levels of props passing (passing props to SettingsTabs and that passing props to each tab) and Filmstrip only reusing the same SettingsDialog as full web.

File: react/features/settings/components/web/SettingsDialog.js [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+import ProfileTab from './ProfileTab';
+import type { Props as ProfileTabProps } from './ProfileTab';
+
+const logger = require('jitsi-meet-logger').getLogger(__filename);
+
+/**
+ * The type of the React {@code Component} props of {@link SettingsDialog}.
+ */
+export type Props = {
+    ...$Exact<DeviceSelectionProps>,
```
[the remark this answers is not in the document]
Comment (saghul) [reply react/features/settings/components/web/SettingsDialog.js]: > I was initially using {...props} for each tab's JSX but was getting worried it was too obtuse but maybe your view is that it's clear enough?

Yep, exactly this :-) But they would be something like `...fooBarProps`, which I think it's pretty clear.

[the remark this answers is not in the document]
Comment (virtuacoplenny) [reply react/features/settings/components/web/SettingsDialog.js]: Ah, so `type Props = { fooProps: Object, barProps: Object, bazProps: Object }`? If I am understanding that correctly, would there not be a risk of unnecessary re-renders potentially due to new objects being created and passed in as props. The perf penalty is probably negligible and I haven't done any measurements to be worried just yet.

[the remark this answers is not in the document]
Comment (saghul) [reply react/features/settings/components/web/SettingsDialog.js]: Yeah, that’s what I had in mind. I haven’t thought of performance though.

**Comment by virtuacoplenny** [comment]
Rebased onto latest master to fix conflicts.

Currently this PR is waiting on Hristo, who wants to try an alternative implementation.

**Comment by virtuacoplenny** [comment]
Superseded by https://github.com/jitsi/jitsi-meet/pull/3121.

**Pull request closed without merging**